prov/mrail/src/mrail_fabric.c

```diff
@@ -82,7 +83,14 @@ int mrail_fabric_open(struct fi_fabric_attr *attr, struct fid_fabric **fabric,
 	if (!mrail_fabric)
 		return -FI_ENOMEM;
 
-	mrail_fabric->info = mrail_get_info_cached(attr->name);
+	ret = mrail_get_info_cached(attr->name, attr->prov_name, &info);
+	if (ret)
+		return ret;
+
+	// We don't want to use a list to open the fabric
+	info->next = NULL;
```

Comment:
I'll add a fi_freeinfo(info->next); before this line.

Reply:
Setting info->next = NULL means, we can open only one fabric corresponding to rail0.

Reply:
Oh, I didn't realize that you were opening a fabric for each struct in the list. When I didn't set info->next to NULL, a lot of things broke, so there's probably a bug somewhere here that I can fish out.